pkg/aws/cloud_config.go

```diff
@@ -25,3 +28,21 @@ func (cfg *CloudConfig) BindFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&cfg.AccountID, flagAWSAccountID, "", "AWS AccountID for the kubernetes cluster")
 	fs.Var(cfg.ThrottleConfig, flagAWSAPIThrottle, "throttle settings for AWS APIs, format: serviceID1:operationRegex1=rate:burst,serviceID2:operationRegex2=rate:burst")
 }
+
+// function to check if aws accountId got converted to scientific notation, and convert back
+// silently log any improperly formatted ids
+func (cfg *CloudConfig) HandleAccountID(log logr.Logger) {
+	properIDMatched, _ := regexp.MatchString("^(\\d{12})$", cfg.AccountID)
+
+	if properIDMatched || cfg.AccountID == "" {
+		return
+	}
+	log.Error(nil, "The following AWS Account ID is not formatted correctly: "+cfg.AccountID)
+
+	scientificMatched, _ := regexp.MatchString("^(\\d[.])(\\d{11})(e\\+11)$", cfg.AccountID)
+	if scientificMatched {
+		cfg.AccountID = cfg.AccountID[0:13]
+		cfg.AccountID = strings.Replace(cfg.AccountID, ".", "", 1)
+		log.Error(nil, "Using the converted AWS Account ID: "+cfg.AccountID)
```

Comment:
nit: `log.Error(nil, fmt.Sprintf("Using the converted AWS Account ID: %s", cfg.AccountID))`

Reply:
fixed thanks Paul